stored_requests/backends/db_fetcher/fetcher.go

```diff
@@ -49,23 +49,31 @@ func (fetcher *dbFetcher) FetchRequests(ctx context.Context, ids []string) (map[
 		}
 		return nil, []error{err}
 	}
-	defer rows.Close()
+	defer func() {
+		if err := rows.Close(); err != nil {
+			glog.Errorf("error closing DB connection: %v", err)
+		}
+	}()
 
 	reqData := make(map[string]json.RawMessage, len(ids))
-	var errs []error = nil
 	for rows.Next() {
 		var id string
 		var thisReqData []byte
+
+		// Fixes #338?
 		if err := rows.Scan(&id, &thisReqData); err != nil {
-			errs = append(errs, err)
+			return nil, []error{err}
 		}
 
 		reqData[id] = thisReqData
```

Comment:
Wouldn't we fix the problem if we simply put this line into an `else`, that is only populate `reqData[]` if there were no errors? Or is there some logic downstream that depends on all IDs being populated if we don't return a nil?

Reply:
That was actually my first thought too, but... on closer investigation, I don't think so.

I tried to write a failed test to verify it, but `sqlmock` doesn't let you fake "scan errors". There's a good explanation of why here: https://github.com/DATA-DOG/go-sqlmock/issues/47

I tried to save other types as mock row data, but... it seems like Go is pretty liberal about what it coerces into `[]byte`. Both `true` and `-1` came back as real data, with no `Scan` error. I'm not sure what if there are other types of data which we could use to force it, either.

Judging by the race condition tickets I linked in #338, though, I think this behavior is caused by something much less straightforward.

Reply:
Ahh, yes, dug into them now. If the buffer is being corrupted, that makes subsequent calls suspect. May be able to get away with exiting returning the rows collected prior to the row with the error. But then at that point just tossing everything out may not be a bad response either.

Sounds like we should be able to revisit after go 1.10 is out, if we remember this issue.